### DISCUSSION TITLE
CRIS: author profile and organization profile

### DISCUSSION THREAD

**Discussion by ulysseus-eu** [inveniosoftware/product-rdm#153] [Ideas]
# Problem statement
In the scope of Ulysseus European University we are building a RDM based on Invenio.
We want to extend InvenioRDM so that it becomes our Current Research Information System as well, containing public researcher profiles and public affiliation profiles, the same way DSpace CRIS does it.
ex for Universitad de Sevilla:
- https://idus.us.es/author-profiles/author?id=2
- https://prisma.us.es/investigador/2

## Describe the solution you'd like
We'd like to contribute and build the required modules (person, organization), required workflows:
- claim the ownership of a public person profile
- claim a mention as contributor on a record as a researcher to link it with your public profile
- claim the ownership of a public organization profile
- claim the affiliation of a record as an organization and would like to link it to your institution profile
- put instance admin workflows in the picture
- allow the automation of detection of authors and affiliations through APIs and hooks

## Describe alternatives you've considered
We have already added custom fields and built custom pages that connect to an external database but we feel it's limited and we miss an opportunity to contribute and to make the ecosystem benefit from such a feature.

## Additional context
We (Ulysseus digital team) are open to be the main contributor of this development if we get collaborative guidance on the best ways to integrate these features (vs communities, vs users) and software architecture.
We'd develop in late 2023 and along 2024.

What do you think?

**Comment by ulysseus-eu** [comment]
We have been working on this feature design today to further discuss with the product and architecture teams.
Here are few diagrams attached.
First a class diagram of database objects
![InvenioPublicProfileClassDiagram drawio](https://github.com/inveniosoftware/invenio-app-rdm/assets/103051002/35a1d687-3a3e-4924-b1ee-5ec7392657a2)

Then we thought users would file claims on public profiles or on specific records to add them to their public profile ( records added by a third party in our case).
Those claims would be resolved by global public profile admins.

Here is the sequence diagram for the resolution of claim for a list of records
![GrantClaimGetRecordsForProfile drawio](https://github.com/inveniosoftware/invenio-app-rdm/assets/103051002/e0f156d8-1f2a-4de3-8f15-b58e21977a79)

Here is the sequence diagram for the resolution of a claim to administrate an existing (pre-generated) public profile.
![GrantClaimGetProfileWithOrWithoutReplacement drawio](https://github.com/inveniosoftware/invenio-app-rdm/assets/103051002/29f6b37a-5298-4d84-97bb-a370f1f074f3)

We'll contact you on discord to discuss live and will  update this issue with the conclusions and next steps.

**Comment by ulysseus-eu** [comment]
We have added some more details in our fork of invenio-docs: https://github.com/ulysseus-eu/docs-invenio-rdm/blob/ulysseus/docs/develop/architecture/claim_profile_or_record.md

**Comment by ulysseus-eu** [comment]
While I was investigating, I started to have a look at [invenio-communities](https://github.com/inveniosoftware/invenio-communities) module which seemed to have similar  features.
Doing so, I realized that invenio-communities could even fit the requirement by translating the following concept:
A claim would be a Request
A PublicProfile would be a Community addressing only one researcher
A ProfileAdmin would be a Community member with owner role
ProfileToRecordAttribution object would be the association between records and the PublicProfile community

And the workflows are already existing within community objects:
creating a community => public profile
requesting to be part of a community => claiming a profile
add a record to a community => adding a record to a public profile
etc.

So it seems I could have sort of a backend or a parent class to use.

**Comment by lnielsen** [comment]
Hi @ulysseus-eu,

Thanks a lot for the interest to contribute to InvenioRDM. I think the easiest approach would be to setup a meeting to discuss further. Can you reach out to me on lars.holm.nielsen@cern.ch and we can organise a call.

Indeed requests module can handle quite a lot of all the claim requests. Also, there's Invenio-accounts and invenio-userprofiles who could extended to address some of the use cases. Also Invenio-Vocabularies already has names and affiliations vocabularies that match up with profiles and organisations.

I would probably not use communities as the public profile for a researchers as communities have a lot of features around collaboration (managing members, submission rules to the communities, etc) which would seem quite odd on a researcher profile page. For organisation profiles I think communities could work well.

I can also explain more about how we try to manage the contributions etc.

**Comment by lnielsen** [comment]
FYI: Here's also a very old issues with some ideas for research profiles that never got realised: https://github.com/zenodo/zenodo/issues/1212

[a comment by an automated account is not shown]

**Comment by egabancho** [comment]
We have started working on what we call "Faculty profiles", https://github.com/ubiquitypress/invenio-faculty-profiles
It is rather basic now, but I think we have the same end goal in mind for it.
